17 enforce storage limit

broija opened the pull request:
closes #17

broija committed:
[FEAT][WIP] Storage limit (#17_enforce_storage_limit)
- Storage limit saved.
- Calculating remaining free space after each download.
- Added FreeSpaceExhausted component.

broija committed:
[FEAT] Storage limit (#17_enforce_storage_limit)
- Added user config.
- Added IPC message to check free space.
- Storage selection : reduce custom selection button sizes.
- FreeSpaceExhausted component :
  - Added buttons.
  - Turned into success component.
- DispatchRequestPayload : node_id is now a number.
- Dispatcher : free space sent is now remaining allocated space.

kacime commented on src/preload/index.ts:
```diff
@@ -72,12 +72,22 @@ const API = {
+  // Save prefs
+  setStorageAllocation: (storageBytes: number) => 
+    ipcRenderer.invoke('set-storage-allocation', storageBytes),
   setBandwidthAllocation: (percentage: number) => 
```
des méthodes ne servent plus à rien visiblement

broija replied:
- setBandwidthAllocation pourrait resservir dans un avenir plus ou moins proche : limitation de la bande passante.
- setStorageAllocation est utilisée.

kacime commented on src/lib/electron-app/utils/user-config.ts:
```diff
@@ -0,0 +1,53 @@
+        this.setValue('storageAllocation', value)
+    }
+
+    public getBandwidthAllocation() {
```
pas utilisé ?

broija replied:
C'est prévu.

kacime commented on src/lib/electron-app/utils/user-config.ts:
```diff
@@ -0,0 +1,53 @@
+        return this.getValue('bandwidthAllocation')
+    }
+
+    public setBandwidthAllocation(value: number) {
```
pas utilisé ?

broija replied:
Pas pour l'instant, mais c'est prévu.

kacime commented on src/lib/electron-app/factories/ipcs/register-system-bandwidth-info.ts:
```diff
@@ -172,28 +173,9 @@ export function registerSystemInfo() {
+    userConfig.setStorageAllocation(storageBytes);
   })
 
   ipcMain.handle('set-bandwidth-allocation', async (event, percentage: number) => {
```
plus utilisé ?

broija replied:
Nettoyé

kacime commented on src/lib/electron-app/utils/user-config.ts:
```diff
@@ -0,0 +1,53 @@
+    private setValue(key: string, value: number) {
+        try {
+            const configPath = this.configPath;
+            const config = JSON.parse(fs.readFileSync(configPath, 'utf8'))
```
ici ça casse si le fichier n'existe pas

il manque la logique de ton if

if (fs.existsSync(configPath)) {
                    try {
                      config = JSON.parse(fs.readFileSync(configPath, 'utf8'))
                    } catch (e) {
                      config = {}
                    }
                  }

kacime reviewed (changes requested):
Quelques commentaires pour un faire un peu de cleaning pour ce qui ne sert plus + il manque un petit bout de code pour créer le fichier .shelter-config.json si il n'existe pas

broija committed:
[FIX] (#17_enforce_storage_limit)
- UserConfig : create ~/.shelter-config.json when necessary.
- Simplified registerSystemInfo for bandwith limitation.
